store/mockstore/mocktikv/cluster.go

```diff
@@ -285,14 +285,22 @@ func (c *Cluster) ScanRegions(key []byte, limit int) ([]*metapb.Region, []*metap
 	})
 
 	keyLocation := sort.Search(len(regions), func(i int) bool {
-		endKey := regions[i].Meta.GetEndKey()
-		if len(endKey) == 0 {
+		if len(regions[i].Meta.GetEndKey()) == 0 {
 			return true
 		}
-		return bytes.Compare(regions[i].Meta.GetEndKey(), key) > 0
+		return bytes.Compare(regions[i].Meta.GetEndKey(), startKey) > 0
 	})
 	regions = regions[keyLocation:]
-	if len(regions) > limit {
+	if len(endKey) > 0 {
+		var pos int
+		for pos = 1; pos < len(regions); pos++ {
```

Comment:
why not use binary search?

Reply:
Done.
Although I change it to binary search, I still don't think this change is meaningful.
There is a `sort.Slice` function, and it's a much heavy operation then search.